ddtrace/tracer/abandoned_spans.go

```diff
@@ -0,0 +1,224 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package tracer
+
+import (
+	"container/list"
+	"fmt"
+	"strings"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+var tickerInterval = time.Minute
+var logSize = 9000
+
+// isBucketNode takes in a list.Element and checks if it is a nonempty
+// list Element
+func isBucketNode(e *list.Element) (*list.List, bool) {
```

Comment:
I'm wondering if there's a slightly better name for these two functions since typically I think of functions named `is_` as just returning a simple boolean and not returning a converted value? Maybe something like `tryAsBucketNode` (not sure I love that name either though). Definitely worth mentioning this behavior in the comment too